### DISCUSSION TITLE
add siteHandle as graphql field

### DISCUSSION THREAD

**Discussion by jan-thoma** [craftcms/cms#10829] [Ideas] [answered]
With graphQL it's possible to narrow the results per site by it's handle, but not to retrieve the sites handle inside the query. Support for siteHandle would be great.

At the moment it's only possible to query an entries siteId like:

```
{
  entries(site: "siteHandle1") {
    siteId
  }
}
```

It would be great to have the possibility to query the siteHandle like:

```
{
  entries(site: "siteHandle1") {
    siteId
    siteHandle
  }
}
```

**Comment by bsye** [comment]
Up!

I'm also interested in this.
What I find odd is that the `siteId` isn't as meaningful as the site handle, and you can't see the ID of a site unless you get it from the admin url (for example: /admin/settings/sites/1?site=siteEn).

Also, since the `siteId` is generated, it isn't a good idea to use it in the frontend

**Comment by davidhellmann** [comment]
Totally +1 here! 
siteId
siteHandle
siteLanguage

**Reply by brianjhanson** [reply]
@davidhellmann just want to call out that you can already get `siteLanguage` by using the `language` field on any element. That will return the language of the site element is associated with.

**Reply by davidhellmann** [reply]
Thanks, good to know!

**Comment by brandonkelly** [comment, marked as answer]
`siteHandle` has been added for Craft 4.3 via #12085 🎉

**Reply by brandonkelly** [reply]
Craft 4.3 was released earlier this week with this change!
